core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -565,8 +560,8 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     val (libraryAttributeNames, workspaceAttributeNames) = destWorkspaceRequest.attributes.keys.partition(name => name.namespace == AttributeName.libraryNamespace)
     withAttributeNamespaceCheck(workspaceAttributeNames) {
       withLibraryAttributeNamespaceCheck(libraryAttributeNames) {
-        dataSource.inTransaction({ dataAccess =>
-          withWorkspaceContextAndPermissions(sourceWorkspaceName, SamWorkspaceActions.read, dataAccess) { sourceWorkspaceContext =>
+        withWorkspaceContextAndPermissionsF(sourceWorkspaceName, SamWorkspaceActions.read) { sourceWorkspaceContext =>
+          dataSource.inTransaction({ dataAccess =>
```

Comment:
this is the general pattern in this PR: switch to the Future-ified version `withWorkspaceContextAndPermissionsF`,  then swap the ordering of `withWorkspaceContextAndPermissionsF` and  `dataSource.inTransaction` in the calling code.